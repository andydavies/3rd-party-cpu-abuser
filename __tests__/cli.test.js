const cpuAbuser = require('../index')
const mock = 'samples/VA-article.json'
const cmd = require('node-cmd')

jasmine.DEFAULT_TIMEOUT_INTERVAL = 15000

describe('Cli executable output', () => {
  it('should return the same output', () => {
    // http://riaevangelist.github.io/node-cmd/
    return new Promise( (resolve, reject) => {
      cmd.get('node cli.js '+mock, (output) => {
        expect(output).toMatchSnapshot()
        return resolve()
      })
    })
  })
})
