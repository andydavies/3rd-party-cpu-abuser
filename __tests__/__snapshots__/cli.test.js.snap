exports[`Cli executable output should return the same output 1`] = `
"Analyzing samples/VA-article.json
┌──────────────────────────┬──────────┐
│ Total CPU busy time (ms) │ 18783.10 │
├──────────────────────────┼──────────┤
│ Total number of domains  │ 51       │
├──────────────────────────┼──────────┤
│ Number of big offenders  │ 20       │
└──────────────────────────┴──────────┘
┌───────────────┬───────────────────────────────────────────────────┐
│ CPU Time (ms) │ domain name                                       │
├───────────────┼───────────────────────────────────────────────────┤
│ 3756.77       │ cdn.taboola.com                                   │
├───────────────┼───────────────────────────────────────────────────┤
│ 3712.06       │ votreargent.lexpress.fr.running.cloud.ger.systems │
├───────────────┼───────────────────────────────────────────────────┤
│ 2898.40       │ cdn.teads.tv                                      │
├───────────────┼───────────────────────────────────────────────────┤
│ 1744.12       │ pagead2.googlesyndication.com                     │
├───────────────┼───────────────────────────────────────────────────┤
│ 729.09        │ acdn.adnxs.com                                    │
├───────────────┼───────────────────────────────────────────────────┤
│ 665.84        │ tpc.googlesyndication.com                         │
├───────────────┼───────────────────────────────────────────────────┤
│ 572.76        │ cdn.adnxs.com                                     │
├───────────────┼───────────────────────────────────────────────────┤
│ 542.82        │ s.skimresources.com                               │
├───────────────┼───────────────────────────────────────────────────┤
│ 495.88        │ cas.nl.eu.criteo.com                              │
├───────────────┼───────────────────────────────────────────────────┤
│ 450.12        │ www.google-analytics.com                          │
├───────────────┼───────────────────────────────────────────────────┤
│ 394.92        │ static.lexpress.fr                                │
├───────────────┼───────────────────────────────────────────────────┤
│ 379.92        │ www6.smartadserver.com                            │
├───────────────┼───────────────────────────────────────────────────┤
│ 306.85        │ ajax.googleapis.com                               │
├───────────────┼───────────────────────────────────────────────────┤
│ 282.58        │ cstatic.weborama.fr                               │
├───────────────┼───────────────────────────────────────────────────┤
│ 185.69        │ dmp.theadex.com                                   │
├───────────────┼───────────────────────────────────────────────────┤
│ 184.69        │ foglio.basilic.io                                 │
├───────────────┼───────────────────────────────────────────────────┤
│ 178.53        │ ib.adnxs.com                                      │
├───────────────┼───────────────────────────────────────────────────┤
│ 167.54        │ ak-ns.sascdn.com                                  │
├───────────────┼───────────────────────────────────────────────────┤
│ 166.27        │ s3.amazonaws.com                                  │
├───────────────┼───────────────────────────────────────────────────┤
│ 164.74        │ js.moatads.com                                    │
└───────────────┴───────────────────────────────────────────────────┘
"
`;
